app/models/marker_assay.rb

```diff
@@ -1,21 +1,20 @@
 class MarkerAssay < ActiveRecord::Base
 
-  belongs_to :marker_sequence_assignment, foreign_key: 'canonical_marker_name'
+  belongs_to :marker_sequence_assignment
 
-  # belongs_to :restriction_enzyme_A, class_name: 'RestrictionEnzyme',
-  #            foreign_key: 'restriction_enzyme_A'
-  # belongs_to :restriction_enzyme_B, class_name: 'RestrictionEnzyme',
-  #            foreign_key: 'restriction_enzyme_B'
+  belongs_to :restriction_enzyme_A, class_name: 'RestrictionEnzyme',
+              foreign_key: 'restriction_enzyme_a_id'
+  belongs_to :restriction_enzyme_B, class_name: 'RestrictionEnzyme',
+              foreign_key: 'restriction_enzyme_b_id'
 
   belongs_to :primer_A, class_name: 'Primer',
-             foreign_key: 'primer_a'
+             foreign_key: 'primer_a_id'
   belongs_to :primer_B, class_name: 'Primer',
-             foreign_key: 'primer_b'
+             foreign_key: 'primer_b_id'
 
-  belongs_to :probe, foreign_key: 'probe_name'
+  belongs_to :probe
 
-  has_many :population_loci, class_name: 'PopulationLocus',
-           foreign_key: 'marker_assay_name'
+  has_many :population_loci, class_name: 'PopulationLocus'
```

Comment:
I have added an inflection initializer in my latest PR, solving the loci - locus problem. Please check if that class_name declaration is still needed.